CollectAudioLogs/CollectAudioLogs.ps1

```diff
@@ -1,4 +1,87 @@
 ﻿# Copyright (c) Microsoft Corporation. All rights reserved.
+param(
+    [string]$problem,
+    [switch]$reproLogsOnly
+);
+
+Function Get-YesNoResponse
+{
+    param([string]$prompt);
+    do 
+    {
+        $response = Read-Host "${prompt}? (y/n)";
+        $response = $response.Trim().ToLower();
+    }
+    while (-not $response.StartsWith("y") -and -not $response.StartsWith("n"));
+    
+    Return $response.StartsWith("y");
+}
+
+[Nullable[bool]]$canRepro;
+[Nullable[bool]]$glitch;
+[bool]$ttt = $false;
+
+switch ($problem)
+{
+    ""
+    {
+    }
+
+    "glitch"
+    {
+        $canRepro = $true;
+        $glitch = $true;
+        break;
+    }
+
+    "norepro"
+    {
+        $canRepro = $false;
+        break;
+    }
+
+    "other"
+    {
+        $canRepro = $true;
+        $glitch = $false;
+        break;
+    }
+
+    "ttt"
+    {
+        $canRepro = $true;
+        $glitch = $false;
+        $ttt = $true;
+        break;
+    }
+
+    default
+    {
+        Write-Host "Unexpected problem $problem - specify 'glitch', 'norepro', or 'other'";
+        Exit;
+    }
+}
+
+if ($null -eq $canRepro)
+{
+    # Ask the user if they'd like to take a repro trace
+    $canRepro = Get-YesNoResponse -prompt "Do you want to grab logs of the problem in action";
+}
+
+if ($canRepro)

```

Comment:
No prompt for TTT if they don't explicitly run the ttt cmd? We've gotten many actionable bugs by prompting for a TTT in the existing logs, i believe it has been very beneficial. I think if they say they can repro and that it's not a problem with audio sounding bad, we should ask if it reproduces across reboot and suggest a TTT.

Reply:
Modified prompt logic